src/amberscript/parser.cc

```diff
@@ -475,6 +475,15 @@ Result Parser::ParseShaderBlock() {
     shader->SetFormat(kShaderFormatSpirvAsm);
     shader->SetData(kPassThroughShader);
 
+    token = tokenizer_->PeekNextToken();
+    if (token->IsIdentifier() && token->AsString() == "TARGET_ENV") {
+      tokenizer_->NextToken();
+      token = tokenizer_->NextToken();
+      if (!token->IsIdentifier() && !token->IsString())
+        return Result("expected target environment after TARGET_ENV");
+      shader->SetTargetEnv(token->AsString());
+    }
+
```

Comment:
Actually, I think we should not allow TARGET_ENV when using a PASSTHROUGH shader. Instead, we should call shader->SetTargetEnv(...) and hard code the target environment, probably to "spv1.0" (because that is what the kPassThroughShader text represents). It turns out that mixing SPIR-V versions in a single pipeline IS allowed, so this should be fine.

Reply:
done